helpers/trees/balance-tree.ts

```diff
@@ -1,14 +1,32 @@
-import MerkleTree from './merkle-tree';
+import { MerkleTree } from './merkle-tree';
 import { BigNumber, utils } from 'ethers';
 
-export default class BalanceTree {
+// Forked from Uniswap: https://github.com/Uniswap/merkle-distributor
+
+export class BalanceTree {
   private readonly tree: MerkleTree;
-  constructor(balances: { account: string; amount: BigNumber }[]) {
-    this.tree = new MerkleTree(
-      balances.map(({ account, amount }, index) => {
-        return BalanceTree.toNode(index, account, amount);
-      })
-    );
+  constructor(
+    balances: { account: string; amount: BigNumber; index?: number }[],
+    customIndex = false
+  ) {
+    if (customIndex) {
```

Comment:
nit: isn't it easier to remove `customIndex` and just check `if (typeof index !== 'undefined')` ?
up to you.

Reply:
unless you can't afford confusion here

Reply:
Good point. `customIndex` needs to be explicitly defined and if one of the items does not have the `index` it must throw an error, otherwise it will output an invalid Merkle root. But, on the other hand, you may want a case where `index` is generated "on the fly" as in the else branch.